Core/src/IfcGeometryConverter/SolidModelConverter.h

```diff
@@ -1722,21 +1730,21 @@ namespace OpenInfraPlatform
 					std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcBlock>(csgPrimitive);
 				if (block)
 				{
-					double x_length = length_factor;
-					double y_length = length_factor;
-					double z_length = length_factor;
+					double x_length = UnitConvert()->getLengthInMeterFactor();
+					double y_length = UnitConvert()->getLengthInMeterFactor();
+					double z_length = UnitConvert()->getLengthInMeterFactor();
 
 					if ((typename IfcEntityTypesT::IfcLengthMeasure)block->XLength)

```

Comment:
I believe these `->XLength`... are not optional - so no need to check for their existence.

Reply:
I have't come so far (I was changing only length_factor to  UnitConvert()->getLengthInMeterFactor() in whole file). I am almost there, so I will correct it as soon as I am there.

Reply:
Right, understood!

Reply:
Resolved: 49fa394